### ISSUE TITLE
Consider Updating CTX Calibration Files to Use New, Frown Removal Files as Default

### ISSUE THREAD

**Issue by jlaura** [DOI-USGS/ISIS3#5279]
**Description**  

A new calibration method for correcting the smile/frown effect in CTX images has been developed by [Walter, Aye, and Munteanu](https://www.hou.usra.edu/meetings/planetdata2023/pdf/7017.pdf). That team has developed new flat field files for CTX. This request would integrate those flat field files into ISIS.

The current method to use the flat files is to manually apply the files during the [execution of `ctxcal`](https://refubium.fu-berlin.de/handle/fub188/37522) (see `proc_ctx.sh`). This request would make the newly released files the default.

@michaelaye Are you planning to continue creating and releasing these or should the ISIS project consider creating and releasing these with new CTX releases?

**Example**  

See the linked abstract above for an example of the processing being performed to create the calibration files (`makeflat`).

**Comment by swalterfub** [comment]
In the long run, it would certainly make sense to integrate the calibration pipeline into ISIS somehow to reduce dependencies.

For the next few years, however, we are confident that we will be able to produce regular updates. Here is a repository where we regularly put the calibration files before they make it into a new release (which includes a DOI update):
[https://git.imp.fu-berlin.de/swalter/ctx-calibration-repository](https://git.imp.fu-berlin.de/swalter/ctx-calibration-repository)

Using the new files with `ctxcal` by default would be great!

**Comment by astrostu** [comment]
As has been raised a few times, to support backwards compatibility, I don't think a new feature should be made the default when it will significantly change the outcome of an ISIS program.  It should certainly be an option, but having it as the default could break others' workflow.

**Comment by jlaura** [comment]
@astrostu I appreciate that this might break existing workflows and so, would suggest that this occur during a major version tick of the software (not that this is covered by the [API definition](https://github.com/DOI-USGS/ISIS3/blob/dev/APIdefinition.md), but as a nicety to the community.)

Other than this being a change to some user's workflows, why wouldn't one want to use better calibration files for the data? For users new to these data, the smile/frown can be concerning and correcting it is a terrific improvement.

**Comment by michaelaye** [comment]
Yes and this is common strategy for other instruments as well so that all new users always get the best available calibration, eg. HiRISE. Is it possible to easily enable to get back to the original calib for a specialist user?

**Comment by swalterfub** [comment]
> Is it possible to easily enable to get back to the original calib for a specialist user?

You just need to provide the "original" calibration file as a parameter on the command line (currently, it's the other way around). The old file just needs to be available.

**Comment by acpaquette** [comment]
@swalterfub Initial integration of the calibration files has worked well. A question from the team about the calibration files, would it be possible for you to host the calibration files via an rclone'able source? It would make it easier on our end to keep the isis data up to date

**Comment by swalterfub** [comment]
Over the past few weeks I have done additional research and it looks like most of the problematic images can be corrected with a single updated calibration file. It looks like only the years 2019 to March 2020 will require a different calibration and I am still investigating why. Apologies if this has caused any unnecessary work!

**Comment by swalterfub** [comment]
The results will be published soon, but this is the current calibration file, which can be integrated by just adding it to `$ISISDATA/mro/calibration`
[ctxFlat_0003.cub.gz](https://github.com/DOI-USGS/ISIS3/files/13473824/ctxFlat_0003.cub.gz)
There should be no visual difference compared to using the monthly calibration files.

**Comment by acpaquette** [comment]
@swalterfub Thank you for providing the file. On testing it looks like there are some differences that we may need to address 
<img width="1616" alt="Screenshot 2023-11-27 at 11 13 42 AM" src="https://github.com/DOI-USGS/ISIS3/assets/17728237/cd06215b-020e-4566-bbd2-6d98fd532ca8">

(Left is monthly, middle is current ctxcal, right is the new 0003 file)

Looking over the cubes in qview with a high contract it looks like the new calibration file has some issues with destrping. The image I am using has a production creation year of 2009 and a start year of 2009 which seems to out of the problem window you mention. Let me know if you need more information on the files I am using.

**Comment by swalterfub** [comment]
@acpaquette Thank you for looking into this - this was actually an intended change of design of the flat-field. The striping should be gone after a subsequent `ctxevenodd` call - could you please confirm this?
In this version we have not included the even/odd channel differences into the flat-field. If you prefer the "old" behavior comparable to the monthly flat-fields and to the 0002 calibration version, I attach the file here (calling it 0004 for now).
Still, I'm wondering that actually I have not seen a real even/odd striping effect in any ISIS-calibrated CTX image yet - and I assume that this has always been included in the flat-fielding already.
 Maybe we should continue discussing whether the flat-field files should be created with eo-corrected images or not.
[ctxFlat_0004.cub.gz](https://github.com/DOI-USGS/ISIS3/files/13505100/ctxFlat_0004.cub.gz)

**Comment by swalterfub** [comment]
I have looked into this a bit more, and can conclude that although the difference is not large, we should stay with the file I've attached as version 4. The pixel-to-pixel variations are a bit less, and the flat-field file is better comparable to the previous one, as this already removed the even/odd variations by the flat-field.
Here is a figure showing the differences:
![with-eo-without-eo](https://github.com/DOI-USGS/ISIS3/assets/17708084/dd39fb47-d651-41b9-9b9f-a23c54ecaf1f)
It shows the sample averages over all lines of 100 randomly chosen images of the whole mission. The red curve shows less even/odd variances and has been calculated using the version 0004 from above.

**Comment by acpaquette** [comment]
@jlaura @michaelaye I think given that this is just a calibration file update from 3 to 4 this can just be added to the data area with no PR/code merge necessary

**Comment by jlaura** [comment]
@acpaquette Will the default behavior be to use this file? That's my requested enhancement as the results using this file are better than the results using the current default. On the documentation side, an update to describe this file and how to use the old capability would be needed as well I think?

**Comment by Kelvinrr** [comment]
@jlaura if we just add the new calibration file to ISISDATA, then yes, adding the file to ISISDATA is all that you need to do as ctxcal looks for files in the format `$mro/calibration/ctxFlat_????.cub` and grabs the highest version. Although, this doesn't solve the problem of giving users the option to opt-in to the new format and would affect users output by default. 

Seems like `ctxFlat_0004.cub` and `ctxFlat_0003.cub` were pushed to ISISDATA on Dec 22nd? Seems like we want to revert this change until a major version bump + docs and/or new parameter to set version defaulting to the old flat file + docs.

**Comment by swalterfub** [comment]
Two separate new files would be confusing in my opinion and I would therefore recommend to delete the file `ctxFlat_0003.cub` and rename the file `ctxFlat_0004.cub` to `ctxFlat_0003.cub`. There aren't any remaining reasons for keeping the intermediate version once this ticket is closed.
Today we have submitted our paper describing the calibration to Earth and Space Science.
I hope it will come out as a pre-print quickly and can then serve as the documentation.
As an addition I can provide the list of images used for the calibration file which should be added to the calibration directory together with a link to the publication or the data repository (in preparation).
To use the previous calibration for backwards compatibility, the users can just provide the "flatfile" parameter to `ctxcal`, e.g.
```
ctxcal from=file.cub to=file.cal.cub flatfile=$ISISDATA/mro/calibration/ctxFlat_0002.cub
```

**Comment by swalterfub** [comment]
Here is the preprint: [10.22541/au.170446769.92634771/v1](https://doi.org/10.22541/au.170446769.92634771/v1)

**Comment by acpaquette** [comment]
@Kelvinrr I added the files on the 22nd. In the past we have been pretty laissez-faire about updating them. I have since removed them and will wait for 9.0

@swalterfub While I agree that a user could do that it breaks the default behavior which is not great under semantic versioning. We could stage the new flat file for access but avoid picking it by default

**Comment by swalterfub** [comment]
The article has now been published: [https://doi.org/10.1029/2023EA003491](https://doi.org/10.1029/2023EA003491)
@acpaquette, could you please only take the version 0004 file and place it as version 0003 in the calibration folder? This is the way I described it in the paper. I don't think we need the one named 0003 from this issue. Thanks!

**Comment by Kelvinrr** [comment]
Closed by #5563
